### PR TITLE
Resoves Issue 2720 - add "search by tag" to contributors index

### PR DESCRIPTION
**Pull request by rtrunkhill**
Added a search function to /app/views/contributors-index.html.erb with autofill script.  

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x ] code is in uniquely-named feature branch and has no merge conflicts
* [ x] PR is descriptively titled
* [ x] PR body includes `fixes #0000`-style reference to original issue #
* [ x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

I see your note in #2720 with screenshot:

image


Looks good -- I think @SidharthBansal said it'd be fine to remove one of the duplicated sections. Leaving some other notes too...

File: app/controllers/wiki_controller.rb [hunk 1]
```diff
@@ -119,10 +119,10 @@ def edit
 
   def new
     @node = Node.new
-    if params[:n] && !params[:body] # use another node body as a template
-      node = Node.find(params[:n])
-      params[:body] = node.latest.body if node&.latest
-    end
+    # if params[:n] && !params[:body] # use another node body as a template
+    #   node = Node.find(params[:n])
+    #   params[:body] = node.latest.body if node&.latest
```
Comment (jywarren) [inline app/controllers/wiki_controller.rb]: see how this uses `node.latest.body` but the lower one uses `node.body` ? I think the top one is strictly more accurate, with `latest`.

File: app/controllers/wiki_controller.rb [hunk 2]
```diff
@@ -136,7 +136,16 @@ def new
...
-    render template: 'wiki/edit'
+    flash.now[:notice] = "This is the new rich editor. For the legacy editor, <a href='/post?#{request.env['QUERY_STRING']}&legacy=true' class='legacy-button'>click here</a>." 
+    if params[:main_image] && Image.find_by(id: params[:main_image]) 
+      @main_image = Image.find_by(id: params[:main_image]).path 
+    end 
+    if params[:n] && !params[:body] # use another node body as a template 
+      node = Node.find(params[:n]) 
+      params[:body] = node.body if node 
+    end 
+    render template: 'editor/wikiRich'
```
Comment (jywarren) [inline app/controllers/wiki_controller.rb]: I think we're trying to solve 2 things at once in this PR - can this be linked back to the request for the rich wiki template as well?

File: test/functional/wiki_controller_test.rb [hunk 3]
```diff
@@ -434,7 +434,8 @@ def teardown
     UserSession.create(users(:jeff))
     get :show, params: { id: 'A-new-wiki-page' }
     assert_response :success
-    assert_template 'wiki/edit'
+    assert_template 'editor/wikiRich'
```
Comment (jywarren) [inline test/functional/wiki_controller_test.rb]: This looks right, and we should be able to remove all the commented lines now. 

Finally, the last error is probably due to the new template having a different tag input form. We should ensure that the test looks for a tag input form that's really there -- it's not finding one with id `taginput` and value of `one,two`. Hope that helps!

```
test_should_use_existing_node_body_as_template_in_post_form_based_on_param_‘n’(Minitest::Result) [/app/test/functional/wiki_controller_test.rb:51]: Expected at least 1 element matching “#taginput[value=”one,two”]”, found 0.. Expected 0 to be >= 1.
```

**Comment by SidharthBansal** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by rtrunkhill** [comment]
oh sorry, i thought they were resolved.  i havnt looked at them for a long
time, just started a new job. I'll get back to helping out when i get on a
regular schedule

**Comment by SidharthBansal** [comment]
Ok Cool. Feel free to reopen it once you start working on it
Thanks

**Pull request closed without merging**